src/Microsoft.AspNetCore.Razor.Tools/ServerCommand.cs

```diff
@@ -133,7 +149,26 @@ internal FileStream WritePidFile()
             var processId = Process.GetCurrentProcess().Id;
             var fileName = $"rzc-{processId}";
 
-            var path = Environment.GetEnvironmentVariable("DOTNET_BUILD_PIDFILE_DIRECTORY");
+            // Make sure the directory exists.
+            Directory.CreateDirectory(directoryPath);
+
+            var path = Path.Combine(directoryPath, fileName);
+            var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write, FileShare.Read, DefaultBufferSize, FileOptions.DeleteOnClose);
+
+            using (var writer = new StreamWriter(fileStream, Encoding.UTF8, DefaultBufferSize, leaveOpen: true))
+            {
+                var rzcPath = Assembly.GetExecutingAssembly().Location;
+                var content = $"{processId}{Environment.NewLine}rzc{Environment.NewLine}{rzcPath}{Environment.NewLine}{Pipe.Value()}";
+                writer.Write(content);
+            }
+
+            return fileStream;
+        }
+
+        // Internal for testing.
+        internal virtual string GetPidFilePath(Func<string, string> getEnvironmentVariable)
+        {
+            var path = getEnvironmentVariable?.Invoke("DOTNET_BUILD_PIDFILE_DIRECTORY");
```

Comment:
Why would `getEnvironmentVariable` be null?

Reply:
It won't. Just being defensive. Better than letting it throw.

Reply:
Nah